File: chatter_backend/post/urls.py
from django.urls import path
from . import api

urlpatterns = [
    path('', api.get_post_list, name='get_post_list'),
    path('create/', api.create_post, name='create_post'),
    path('<uuid:id>/like/', api.create_like_for_post, name="create_like_for_post"),
    path('<uuid:id>/', api.get_post_details, name="get_post_details"),
    path('<uuid:id>/delete/', api.delete_post, name="delete_post"),
    path('<uuid:id>/comments/create/', api.create_comment_on_post, name='create_comment_on_post'),
    path('<uuid:post_id>/comments/<uuid:comment_id>/delete/', api.delete_comment_on_post, name='delete_comment_on_post'),
    path('<uuid:id>/comments/', api.get_post_comments_list,
         name='get_post_comments_list'),
    path('<uuid:post_id>/comments/<uuid:comment_id>/', api.create_like_for_comment,
         name='create_like_for_comment'),
    path('by/<uuid:id>/', api.get_profile_post_list, name='get_profile_post_list')
]

File: chatter_frontend/src/constants/urls.js
const URLS = {
  refreshToken: "/api/refresh/",
  signUp: "/api/signup/",
  login: "/api/login/",
  me: "/api/me/",
  allPosts: "/api/posts/",
  createPost: "/api/posts/create/",
  userPosts: (id) => `/api/posts/by/${id}`,
  search: (query) => `/api/search/${query}`,
  addFriend: (id) => `/api/friends/${id}/add/`,
  getFriendsAndRequests: (id) => `/api/friends/${id}/`,
  respondToFriendRequest: (status, senderId) => `/api/friends/request/${senderId}/${status}/`,
  deleteSentRequest: (receiverId) => `/api/friends/request/${receiverId}/delete/`,
  likePost: (id) => `/api/posts/${id}/like/`,
  createComment: (postId) => `/api/posts/${postId}/comments/create/`,
  getPostCommentsList: (postId) => `/api/posts/${postId}/comments/`,
  likeComment: (postId, commentId) => `/api/posts/${postId}/comments/${commentId}/`,
  deleteComment: (postId, commentId) => `/api/posts/${postId}/comments/${commentId}/delete/`,
  deletePost: (id) => `/api/posts/${id}/delete/`,
  getPost: (id) => `/api/posts/${id}/`,
  getConversationList: "/api/chat/conversations/",
  getConversation: (conversationId) => `/api/chat/conversations/${conversationId}/`,
  sendMessage: (conversationId) => `/api/chat/conversations/${conversationId}/send/`,
  startConversation: (userId) => `/api/chat/conversations/${userId}/start/`,
}

export default URLS